### PR TITLE
Only check permissions

### PR DESCRIPTION
**Pull request by lorentzlasson**
https://github.com/stiftelsen-effekt/admin/issues/24

---

Tested on devices

- [ ] Desktop 💻
- [ ] Mobile 📱

Tests

- [ ] All tests are running ✔️
- [ ] Test are updated 🧪
- [ ] Code Review 👩‍💻
- [ ] QA 👌

⏲️ Time spent on CR:

⏲️ Time spent on QA:

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -61,7 +61,6 @@ module.exports = {
   // Auth0
   authAudience: process.env.AUTH_AUDIENCE,
   authIssuerBaseURL: process.env.AUTH_BASE_URL,
-  authRoleClaim: process.env.AUTH_ROLE_CLAIM,
```
Comment (lorentzlasson) [inline src/config.js:64]: Should not have the need to look at anything other than permissions

File: src/custom_modules/authorization/authMiddleware.ts [hunk 2]
```diff
@@ -1,56 +1,50 @@
...
 const userIdClaim = authUserIdClaim;
 
 /**
  * Express middleware, checks if token passed in request grants permission
  * If api is true, stops the request on fail and sends 401 unathorized
  * If api is false, sets req.authorized to true on success and false on fail, then calls next
  * @param {String} permission Shortname permission
- * @param {Boolean} [api=true] Indicates whether the request is an api request or a view request
  */
-export const auth = (permission, api = true) => {
```
Comment (lorentzlasson) [inline src/custom_modules/authorization/authMiddleware.ts:29]: Unused

File: src/custom_modules/authorization/authMiddleware.ts [hunk 3]
```diff
@@ -1,56 +1,50 @@
...
+export const auth = (permission: any) => {
   return [checkJwt, claimIncludes("permissions", permission)];
 };
 
-const userIsAdmin = (claims: JWTPayload) => {
-  const roleClaims = claims[roleClaim];
-  if (!Array.isArray(roleClaims)) return false;
-  return roleClaims.includes("admin");
+export const isAdmin = auth(authorizationPermissions.admin);
```
Comment (lorentzlasson) [inline src/custom_modules/authorization/authMiddleware.ts:24]: To clarify it's a super thin layer on top of `auth`

File: src/enums/authorizationPermissions.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-module.exports = {
+export default {
```
Comment (lorentzlasson) [inline src/enums/authorizationPermissions.js:1]: Renaming file as this is a map/enum of permissions, not roles

File: src/custom_modules/authorization/authMiddleware.ts [hunk 5]
```diff
@@ -1,56 +1,50 @@
...
+
+const userIsAdmin = (jwtPayload: JWTPayload) => {
+  const permissionJwtPayload = jwtPayload.permissions;
```
Comment (lorentzlasson) [inline src/custom_modules/authorization/authMiddleware.ts:27]: The admin role contains the `admin` permission so this should be enough and streamline how we handle authorization

File: src/routes/vipps.ts [hunk 6]
```diff
@@ -152,54 +152,46 @@ router.get("/histogram/charges", async (req, res, next) => {
...
+    let content = await DAO.vipps.getAgreementReport();
+
+    res.json({
+      status: 200,
+      content,
+    });
+  } catch (ex) {
+    next(ex);
+  }
+});
```
Comment (lorentzlasson) [inline src/routes/vipps.ts:166]: Just auto formatting shuffling things around

File: src/custom_modules/authorization/authMiddleware.ts (hunk 2 above)
Comment (fellmirr) [reply src/custom_modules/authorization/authMiddleware.ts:29]: Yep, it's an old relic from when we had our custom authorization solution :+1:

**Review by fellmirr — APPROVED** [review]
Good stuff 👍

**Pull request merged**